### PR TITLE
Refs #25974 - add applied errata report

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by ares** [comment]
needs https://github.com/Katello/katello/pull/7958 and https://github.com/theforeman/foreman/pull/6458

**Comment by ares** [comment]
Both dependencies are merged now, this can be easily tested with nightlies now. Could someone please take a look?

**Comment by ares** [comment]
@lhellebr, could you please test on nightlies? just import this template and render it, you need to have some errata applied (ideally both using rex and katello) to see something

File: report_templates/applied_errata.erb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  input_type: user
+  description: Last reboot information is based on configuration management facts,
+    such as uptime reported by Puppet or Ansible. The computation might be performance
+    heavy, if this value is not interesting, it can be omitted by choosing no.
+  options: "yes\r\nno"
+  advanced: false
+model: ReportTemplate
+require:
+- plugin: katello
+￼ version: 3.12.0
```
Comment (lhellebr) [inline report_templates/applied_errata.erb]: The first symbol on this line is fishy

Comment (ares) [reply report_templates/applied_errata.erb]: not sure what happend but it shoud be now two spaces, thanks

**Comment by lhellebr** [comment]
When rendering empty report, it works. After adding some errata aplied through **katello** (without remote execution), the following pop-up message appears:

`uninitialized constant Katello::Concerns::BaseTemplateScopeExtensions::RemoteExecutionFeature Did you mean? Katello::RemoteExecutionController`

With the attached traceback. [res.txt](https://github.com/theforeman/community-templates/files/2941900/res.txt)

**Comment by ares** [comment]
@lhellebr this should be resolved by https://github.com/Katello/katello/pull/8020, turns out the macro in Katello relied on REX plugin installed, which is missing in your instance

**Comment by lhellebr** [comment]
Reproduced right now, with the very machine you provided to me, AND even without any errata.

**Comment by ares** [comment]
With the Katello PR applied? Does that PR introduce a "regression" then?

**Comment by lhellebr** [comment]
That's the issue, I haven't applied the PR. I thought it was in nightly already. So this was a false alarm.
The fact that it happened without any errata seems like a "regression" but I'll retest with the PR applied.

**Review by tbrisker — APPROVED** [review]
fine with getting this in before branching if it gets tested

**Comment by san7ket** [comment]
I am trying to test this, 

Steps: 
1. host with some errata 
2. I applied errata on the host from satellite. The task was successfully
3. Did a errata recalculate on host, now there is no errata
4. On generating the report I see empty report.

What am I doing wrong? I haven't sync report template. Just copied it from here created a new report template with all the needed inputs, does this work ?

**Comment by ares** [comment]
to test:
1) download this file to ~foreman/app/views/unattended/report_templates/
2) you need to edit the file and remove katello 3.12 requirement, it's not released yet, so the template wouldn't be seeded
3) run rake db:seed (foreman-rake db:seed on pro like env)
4) the template should appear in Monitor -> Report Templates -> Applied Errata (not to be confused with Applicable Errata)
5) make sure you deployed some errata using katello-agent, some using REX (regardless of result), the report should show these activities

**Comment by san7ket** [comment]
I tried the steps, I wasn't able to seed it.
After I run the db:seed there was no error, it just skipped creating the Applied errata template, and require katello version was removed. I am using the develop setup

**Comment by san7ket** [comment]
I was able to generate the report successfully, 

```
date,hostname,erratum_id,erratum_type,status,last_reboot_time
2019-04-09 19:40:08 +0530,katello-client.local.example.com,RHEA-2012:0055,security,success,1970-01-01 06:02:56 +0530
```
```
date,hostname,erratum_id,erratum_type,status
2019-04-09 19:47:22 +0530,katello-client.local.example.com,RHEA-2012:0055,security,error
2019-04-09 19:47:58 +0530,katello-client.local.example.com,RHEA-2012:0055,security,error
```

**Review by san7ket — APPROVED** [review]

**Comment by ares** [comment]
Thank you @san7ket, @tbrisker, mind to merge now? Thanks

**Review by tbrisker — APPROVED** [review]
Thanks @ares and @san7ket !

**Pull request merged**